Libraries/LibWeb/CSS/Parser/Helpers.cpp

```diff
@@ -138,4 +140,76 @@ Vector<CSS::Parser::ComponentValue> parse_component_values_list(CSS::Parser::Par
     return CSS::Parser::Parser::create(parsing_params, string).parse_as_list_of_component_values();
 }
 
+// https://drafts.csswg.org/css-syntax/#css-decode-bytes
+ErrorOr<String> css_decode_bytes(Optional<StringView> const& environment_encoding, Optional<String> mime_type_charset, ByteBuffer const& encoded_string)
+{
+    // https://drafts.csswg.org/css-syntax/#determine-the-fallback-encoding
+    auto determine_the_fallback_encoding = [&mime_type_charset, &environment_encoding, &encoded_string]() -> StringView {
+        // 1. If HTTP or equivalent protocol provides an encoding label (e.g. via the charset parameter of the Content-Type header) for the stylesheet,
+        //    get an encoding from encoding label. If that does not return failure, return it.
+        if (mime_type_charset.has_value()) {
+            if (auto encoding = TextCodec::get_standardized_encoding(mime_type_charset.value()); encoding.has_value())
+                return encoding.value();
+        }
+        // 2. Otherwise, check stylesheet’s byte stream. If the first 1024 bytes of the stream begin with the hex sequence
+        // 40 63 68 61 72 73 65 74 20 22 XX* 22 3B
+        // where each XX byte is a value between 0x16 and 0x21 inclusive or a value between 0x23 and 0x7F inclusive, then get an encoding from a string formed out of the sequence of XX bytes, interpreted as ASCII.
+        auto check_stylesheets_byte_stream = [&encoded_string]() -> Optional<StringView> {
+            size_t scan_length = min(encoded_string.size(), 1024);
+            auto pattern_start = "@charset \""sv;
+            auto pattern_end = "\";"sv;
+
+            if (scan_length < pattern_start.length())
+                return {};
+
+            StringView start_view(encoded_string.data(), pattern_start.length());
+            if (start_view != pattern_start)
+                return {};
+
+            auto encoding_start = pattern_start.length();
+            StringView buffer_view(encoded_string.data(), scan_length);
```

Comment:
This feels a bit dangerous, and we have more type-safe ways of doing it:
- `encoded_string.bytes()` returns a Bytes, which is a view onto the data (no copies)
- ...which you can get the first 1024 chars of with `.slice(start, length)`
- ...and then you can construct a `StringView` directly from that.